src/game/savegame.c

```diff
@@ -593,6 +593,12 @@ void Savegame_ScanAvailableLevels(REQUEST_INFO *req)
 
 bool Savegame_RestartAvailable(int32_t slot_num)
 {
+    if (slot_num == -1) {
+        return true;
+    }
+
+    Savegame_ScanSavedGames();
```

Comment:
No. Please refrain from introducing side effects to functions intended to be read-only.

Instead please update GAME_INFO when updating current_save_slot.

Reply:
This function checks the `savegame_info = &m_SavegameInfo[slot_num]` for the restart level flag, so I think we need to update the slot info in some way and not just GAME_INFO. Should I change the function to instead check `GAME_INFO.save_initial_version`? It's a little confusing to have both, so which do you think is clearer to test?

Reply:
Good catch, this should only check for `feature_flags.restart`.

Reply:
Alright I am going to brainstorm on this for a second. So this function `Savegame_RestartAvailable` was made to check if a savegame slot allows restart. However, now we've seen these special cases:
1. New game without any saves or loads and the player dies. The `current_save_slot` is `-1` which technically isn't a save slot though it's like a special new game slot.
2. The player saves over a legacy or empty slot, but the slot doesn't update until `Savegame_ScanSavedGames` is called. Currently, that only happens in `Shell_Main` and `Option_PassportInitSaveRequester`.

So which solution would you prefer?
1. Keep the `current_save_slot` is `-1` check (it is the new game slot). Somewhere `Savegame_ScanSavedGames` would have to be called. Maybe it makes sense to call that at the end of the saving function since that is when the savegame slot is updated? Otherwise we have to do a `Savegame_ScanSavedGames` in the `Option_Passport` function.
2. Not use this `Savegame_RestartAvailable` function (which means not using the savegame slot's info) and instead check what `g_GameInfo.save_initial_version` is. Currently that flag is only used to save the initial version in the BSON savegame's header.
3. Now that I'm looking at it, maybe there's a third option. Maybe there's a way to get rid of `g_GameInfo.save_initial_version` and only use the `SAVEGAME_INFO.initial_version`. We could use the `current_save_slot` is `-1` to determine if the playthrough is a new game aka `SAVEGAME_CURRENT_VERSION`. Might reduce confusion?

Reply:
1)